server/app/controllers/applicant/ApplicantRoutes.java

```diff
@@ -189,16 +189,31 @@ public Call confirmAddress(
   }
 
   /**
-   * Returns the route corresponding to the applicant previous block action.
+   * Returns the route corresponding to the applicant previous block action, or the route
+   * corresponding to the review page if there's no valid previous block.
    *
    * @param profile - Profile corresponding to the logged-in user (applicant or TI).
    * @param applicantId - ID of applicant for whom the action should be performed.
    * @param programId - ID of program to review
-   * @param previousBlockIndex - index of the previous block
+   * @param currentBlockIndex - index of the current block
    * @param inReview - true if executing the review action (as opposed to edit)
    * @return Route for the applicant previous block action
    */
-  public Call blockPrevious(
+  public Call blockPreviousOrReview(
```

Comment:
praise: Kudos for making this logic reusable and breaking it out into its own method.